frontend/src/metabase/visualizer/visualizer.slice.ts

```diff
@@ -324,7 +347,14 @@ const visualizerHistoryItemSlice = createSlice({
       );
 
       if (state.display === "funnel") {
-        addColumnToFunnel(state, column, columnRef, dataset, dataSource);
+        addColumnToFunnel(
+          state,
+          column,
+          columnRef,
+          // Prevents "Type instantiation is excessively deep" error
+          dataset as Dataset,
+          dataSource,
+        );
```

Comment:
Had to cast `dataset: Draft<Dataset>` to `Dataset` to avoid the error. Wrapping `addColumnToFunnel`'s `dataset` argument with `Draft` doesn't help :/